numba/tests/test_dufunc.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import print_function, absolute_import, division
+from numba import unittest_support as unittest
+
+import numpy as np
+
+from numba.npyufunc import dufunc
+
+def pyuadd(a0, a1):
+    return a0 + a1
+
+class TestDUFunc(unittest.TestCase):
+
+    def test_frozen(self):
+        duadd = dufunc.DUFunc(pyuadd, nopython=True)
+        self.assertFalse(duadd.frozen)
+        duadd.frozen = True
```

Comment:
Heh.  I never liked `freeze()` methods that just set a flag, and personally find this cleaner.  Furthermore, this isn't really something intended to be user facing.  This flag should be set by the `vectorize()` decorator when we make DUFunc's the result when signatures are given at decoration time.